kit/src/components/message/mod.rs

```diff
@@ -73,6 +76,10 @@ pub struct Props<'a> {
     parse_markdown: bool,
     // called when a reaction is clicked
     on_click_reaction: EventHandler<'a, String>,
+
+    pending: bool,
+
+    pending_attachments: Option<Vec<(String, Progression)>>,
```

Comment:
the pending field is necessary because even with an empty `pending_attachments` messages can still be pending.

Reply:
I think you're right about `pending` but perhaps for a different reason - I realize now that `kit` can't access state.  I still hope that `pending` can be removed from things like `chat > compose > messages.rs > MessageProps` since the message_uuid and chat_uuid could be used to retrieve that information from state.

Reply:
wouldnt that cause unnecessary lookups instead? as if we already know at message creation that its pending we can just save it there already